src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -1,50 +1,24 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import _ from 'lodash';
 import { useSelector, useDispatch } from 'react-redux';
 import {
   EyeOutlined,
   EyeInvisibleOutlined,
-  UpOutlined,
-  DownOutlined,
 } from '@ant-design/icons';
 import {
-  Button, Space, Switch,
+  Switch,
 } from 'antd';
+
 import { updatePlotConfig } from '../../../../../../redux/actions/componentConfig';
+import ReorderableList from '../../../../../../components/ReorderableList';
 
 const HeatmapMetadataTrackSettings = () => {
   const dispatch = useDispatch();
 
   const cellSets = useSelector((state) => state.cellSets);
-  const selectedTracks = useSelector((state) => state.componentConfig.interactiveHeatmap.config.selectedTracks);
-
-  const [trackData, setTrackData] = useState([]);
-
-  useEffect(() => {
-    setTrackData(
-      _.unionBy(
-        trackData,
-        getCellSets(
-          ['cellSets', 'metadataCategorical'],
-        ).map(
-          (data) => ({ selected: selectedTracks.includes(data.key), key: data.key }),
-        ),
-        'key',
-      ),
-    );
-  }, [cellSets.hierarchy]);
-
-  useEffect(() => {
-    if (trackData.length === 0) {
-      return;
-    }
-
-    dispatch(
-      updatePlotConfig('interactiveHeatmap', {
-        selectedTracks: trackData.filter((o) => o.selected).map((o) => o.key),
-      }),
-    );
-  }, [trackData]);
+  const selectedTracks = useSelector(
+    (state) => state.componentConfig.interactiveHeatmap.config.selectedTracks,
```

Comment:
We plan to use the `HeatmapMetadataTrackSettings` component in other parts of the platform, for instance in plots and tables heatmap in the near future. To make it easy to reuse, I would suggest that you don't hardcode `interactiveHeatmap` in here, but pass the component_type as props. In this way, `HeatmapMetadataTrackSettings` and `HeatmapPlot` are no longer coupled.